enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+    type: array
+    maxItems: 10
+    items:
+        description: AWSResourceTag is a tag to apply to AWS resources created for the cluster.
+        type: object
+        required:
+            - key
+            - value
+        properties:
+            key:
+                description: key is the key of the tag
+                type: string
+                maxLength: 128
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+            value:
+                description: value is the value of the tag. Some AWS service do not support empty values. Since tags are added to resources in many services, the length of the tag value must meet the requirements of all services.
+                type: string
+                maxLength: 256
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+```
+
+### User Stories
 
-If `experimentalPropagateUserTags` is true, install validation will fail if there is any tag that starts with `kubernetes.io` or `openshift.io`.
+- As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
+AWS resource tags.Refer to https://issues.redhat.com/browse/SDE-1146.
 
-Add a new field `resourceTags` to `.status.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
-`resourceTags` field will be applied to new resources created for the cluster. The `resourceTags` field will be populated by the installer only if the `experimentalPropagateUserTags` field is true.
+- As a cluster administrator of OpenShift, I would like to be able to add user-defined tags for AWS resources provisioned by an OCP cluster.
 
-Note existing unchanged behavior: The installer will apply these tags to all AWS resources it creates with terraform (e.g. bootstrap and master EC2 instances) from the install config, not from infrastructure status, and regardless if the propagation option is set.
+- As a cluster administrator of OpenShift, I would like to be able to delete user-defined tags managed by OpenShift for AWS resources.
 
-All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api)
-will apply these tags to all AWS resources they create.
+- As a cluster administrator of OpenShift, I would like to be able to update user-defined tags managed by OpenShift for AWS resources.
 
-userTags that are specified in the infrastructure resource will merge with userTags specified in an individual component. In the case where a userTag is specified in the infrastructure resource and there is a tag with the same key specified in an individual component, the value from the individual component will have precedence and be used.
+- As a cluster administrator of OpenShift, I would like to be able to remove user-defined tags without delete for AWS resources.
```

Comment:
How is this different than the story above about wanting to be able to delete user-defined tags?

Reply:
Rephrased it.